Corona-Warn-App/src/main/java/de/rki/coronawarnapp/notification/NotificationReceiver.kt

```diff
@@ -0,0 +1,51 @@
+package de.rki.coronawarnapp.notification
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import androidx.navigation.NavDeepLinkBuilder
+import de.rki.coronawarnapp.R
+
+import de.rki.coronawarnapp.notification.NotificationConstants.NOTIFICATION_ID
+import de.rki.coronawarnapp.notification.NotificationConstants.POSITIVE_RESULT_NOTIFICATION_ID
+import de.rki.coronawarnapp.storage.LocalData
+import de.rki.coronawarnapp.ui.main.MainActivity
+import timber.log.Timber
+
+typealias NotificationId = Int
+
+class NotificationReceiver : BroadcastReceiver() {
+
+    override fun onReceive(context: Context, intent: Intent) {
+        when (val requestCode = intent.getIntExtra(NOTIFICATION_ID, Int.MIN_VALUE)) {
+            POSITIVE_RESULT_NOTIFICATION_ID ->
+                showPositiveResultNotification(context, requestCode)
+            else ->
+                Timber.tag(TAG).v("NotificationReceiver received an undefined request code: %s", requestCode)
+        }
+    }
+
+    private fun showPositiveResultNotification(context: Context, notificationId: Int) {
+        if (LocalData.numberOfRemainingPositiveTestResultReminders > 0) {
+            LocalData.numberOfRemainingPositiveTestResultReminders -= 1
```

Comment:
As this is the bc receiver class it is only supposed to be the access point for triggers (application layer). I would really love all the reminder logic to be extracted to a class within the domain layer.
An injectable class that could maybe have those members for:
- resetReminding() -> when reminding is needed, because submission was done
- isRemindingNeeded() -> whether or not to display reminders
- onReminderShownOnce() -> when the reminder Is shown

This way we have a single point of responsibility for the reminder logic. I guess testing would also be simplified.